sklearn/datasets/openml.py

```diff
@@ -721,18 +696,20 @@ def fetch_openml(name=None, version='active', data_id=None, data_home=None,
     if not return_sparse:
         # The shape must include the ignored features to keep the right indexes
         # during the arff data conversion.
-        data_qualities = _get_data_qualities(data_id, data_home)
+        data_qualities = memory.cache(_get_data_qualities)(data_id=data_id)
         shape = _get_num_samples(data_qualities), len(features_list)
     else:
         shape = None
 
     # obtain the data
-    arff = _download_data_arff(data_description['file_id'], return_sparse,
-                               data_home, encode_nominal=not as_frame)
-
-    description = "{}\n\nDownloaded from openml.org.".format(
-        data_description.pop('description'))
-
+    kwargs = dict(file_id=file_id, sparse=return_sparse,
+                  encode_nominal=not as_frame)
+    if LooseVersion(joblib.__version__) >= LooseVersion('0.12.0'):
+        arff = memory.cache(_download_data_arff)(**kwargs)
+    else:
+        # in some cases liac-arff returns a generator with data which
+        # cannot be pickled with joblib < 0.12
+        arff = _download_data_arff(**kwargs)
```

Comment:
Is it worth caching this at all? Is it sufficient to cache the full operation without worrying about this – essentially – HTTP caching?